### PR TITLE
Ear 1319 construct importing process

### PR DESCRIPTION
**Pull request by MattElcock**
### What is the context of this PR?

Now that we've created the individual stages of the Importing Content process, this PR stitches them all together.

I've not made any unit tests for the overarching 'handler' file (the one that handles what modal is showing) as I'm going to write some intergration tests within our Cypress library once this has been merged in.

### How to review

Try out the importing process, end to end!

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process

### CODE REVIEW

**Review by ashtuck — DISMISSED** [review]

**Comment by MattElcock** [comment]
After speaking with Sam:

- [ ] Modified date shows 'invalid date'
- [ ] Importing a source questionnaire from 'Editor' pane doesn't work
- [ ] Cannot scroll with lots of questionnaires on 'All' page,
- [ ] 'Showing x of y' not indented
- [ ] Owner on source questionnaire isn't populated
- [ ] When page 1 of 1 the right arrow is clickable when it shouldn't be

**Review by farres1 — CHANGES_REQUESTED** [review]
Requesting changes as I've discussed some bugs with Matt to fix before this ticket is merged.

Matt has a list of these bugs in the above comment.

File: eq-author/src/App/importingContent/index.js [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+        >
+          {({ loading, error, data }) => {
+            if (loading) {
+              return <React.Fragment />;
+            }
+
+            if (error || !data) {
+              return <React.Fragment />;
+            }
+            console.log(data.questionnaires);
```
Comment (farres1) [inline eq-author/src/App/importingContent/index.js:246]: console log

**Review by farres1 — COMMENTED** [review]
Deleting a questionnaire from the home page requires a refresh before it's removed from the table of questionnaires.

File: eq-author/src/components/QuestionnaireTable/index.js [hunk 2]
```diff
@@ -11,6 +11,14 @@ const TableWrapper = styled.table`
   border-collapse: collapse;
   table-layout: fixed;
   text-align: left;
+
+  ${({ variant }) =>
+    variant == "selectModal" &&
```
Comment (farres1) [inline eq-author/src/components/QuestionnaireTable/index.js:16]:
```suggestion
    variant === "selectModal" &&
```

File: eq-author/src/components/QuestionnairesView/PaginationNav/index.js [hunk 3]
```diff
@@ -16,6 +16,12 @@ const Results = styled.div`
   color: ${colors.textLight};
   font-size: 0.9em;
   position: absolute;
+
+  ${({ padding }) =>
+    padding == "small" &&
```
Comment (farres1) [inline eq-author/src/components/QuestionnairesView/PaginationNav/index.js:21]:
```suggestion
    padding === "small" &&
```

File: eq-author/src/components/QuestionnairesView/PaginationNav/index.js [hunk 4]
```diff
@@ -24,9 +30,10 @@ const PaginationNavTable = ({
   pageCount,
   currentPageIndex,
   onPageChange,
+  padding,
```
Comment (farres1) [inline eq-author/src/components/QuestionnairesView/PaginationNav/index.js:33]: `padding` missing from props validation

File: eq-author/src/components/QuestionnaireTable/index.js [hunk 5]
```diff
@@ -19,17 +27,19 @@ const QuestionnaireTable = ({
...
   autoFocusId,
   onDeleteQuestionnaire,
   onDuplicateQuestionnaire,
   handleLock,
   enabledHeadings,
   clickable,
   onRowClick,
-  questionnaireModal
+  questionnaireModal,
+  variant,
```
Comment (farres1) [inline eq-author/src/components/QuestionnaireTable/index.js:39]: `variant` missing from props validation

**Review by farres1 — CHANGES_REQUESTED** [review]

**Review by farres1 — APPROVED** [review]

**Review by zimboflyman — APPROVED** [review]
Monster PR - nice job Matt! 😊

**Pull request merged**